jpegenc/subblocks/dct_1d/dct_1d.py

```diff
@@ -128,23 +128,23 @@ def coeff_assign():
     def mul_add():
         """multiplication and addition"""
         if input_interface.data_valid:
-                mult_reg[0].next = data_in_reg * coeffs[0]
-                mult_reg[1].next = data_in_reg * coeffs[1]
-                mult_reg[2].next = data_in_reg * coeffs[2]
-                mult_reg[3].next = data_in_reg * coeffs[3]
-                mult_reg[4].next = data_in_reg * coeffs[4]
-                mult_reg[5].next = data_in_reg * coeffs[5]
-                mult_reg[6].next = data_in_reg * coeffs[6]
-                mult_reg[7].next = data_in_reg * coeffs[7]
-
-                adder_reg[0].next = mux_flush[0] + mult_reg[0]
-                adder_reg[1].next = mux_flush[1] + mult_reg[1]
-                adder_reg[2].next = mux_flush[2] + mult_reg[2]
-                adder_reg[3].next = mux_flush[3] + mult_reg[3]
-                adder_reg[4].next = mux_flush[4] + mult_reg[4]
-                adder_reg[5].next = mux_flush[5] + mult_reg[5]
-                adder_reg[6].next = mux_flush[6] + mult_reg[6]
-                adder_reg[7].next = mux_flush[7] + mult_reg[7]
+            mult_reg[0].next = data_in_reg * coeffs[0]
```

Comment:
Yes there is a reason. If a for loop is used then none of the Xilinx or Altera synthesizers will recognize the multiplications as DSP units. Instead with this configuration all the multiplications are recognized and assigned to DSP units. Check page 99 of http://www.xilinx.com/itp/xilinx10/books/docs/sim/sim.pdf